File: wiz-event-mall-java/src/main/java/com/example/wizeventmalljava/api/event/EventController.java
package com.example.wizeventmalljava.api.event;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import com.example.wizeventmalljava.api.event.dto.EventCreateRequest;
import com.example.wizeventmalljava.api.event.dto.EventInfoResponse;
import com.example.wizeventmalljava.api.event.dto.EventParticipateResponse;

import lombok.RequiredArgsConstructor;

@RestController
@RequestMapping("/api/events")
@RequiredArgsConstructor
public class EventController {

	private final EventService eventService;

	@PostMapping
	public ResponseEntity<Long> createEvent(@RequestBody EventCreateRequest req) {
		Long eventId = eventService.create(req);
		return ResponseEntity.ok(eventId);
	}

	@GetMapping("/{eventId}")
	public ResponseEntity<EventInfoResponse> retrieveEventInfo(@PathVariable Long eventId) {
		EventInfoResponse response = eventService.retrieve(eventId);
		return ResponseEntity.ok(response);
	}

	@PostMapping("/{eventId}/participate")
	public ResponseEntity<EventParticipateResponse> participateEvent(
		@PathVariable Long eventId,
		@RequestParam Long memberId
	) {
		EventParticipateResponse response = eventService.participate(eventId, memberId);
		return ResponseEntity.ok(response);
	}

	@PostMapping("/{eventId}/participate/v2")
	public ResponseEntity<EventParticipateResponse> participateEventWithLock(
		@PathVariable Long eventId,
		@RequestParam Long memberId
	) {
		EventParticipateResponse response = eventService.participateWithLock(eventId, memberId);
		return ResponseEntity.ok(response);
	}
}

File: wiz-event-mall/src/main/kotlin/com/example/wizeventmall/api/event/EventController.kt
package com.example.wizeventmall.api.event

import com.example.wizeventmall.api.event.dto.EventInfoResponse
import com.example.wizeventmall.api.event.dto.EventParticipateResponse
import com.example.wizeventmall.api.event.dto.EventRegisterRequest
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/v1/events")
class EventController(
    private val eventService: EventService
) {

    @PostMapping
    fun registerEvent(@RequestBody req: EventRegisterRequest): ResponseEntity<Long> {
        return eventService.register(req)
            .let { ResponseEntity.status(HttpStatus.CREATED).body(it) }
    }

    @GetMapping("/{eventId}")
    fun retrieveEventInfo(@PathVariable eventId: Long): ResponseEntity<EventInfoResponse> {
        return eventService.retrieve(eventId)
            .let { ResponseEntity.ok(it) }
    }

    // TODO : 구현 편의성을 위해 memberId 를 파라미터로 받음
    @PostMapping("/{eventId}/participate")
    fun participateEvent(
        @PathVariable eventId: Long,
        @RequestParam memberId: Long,
    ): ResponseEntity<EventParticipateResponse> {
        return eventService.participate(eventId, memberId)
            .let { ResponseEntity.ok(it) }
    }

}